slatedb/src/transactional_object/mod.rs

```diff
@@ -0,0 +1,919 @@
+//! SlateDB Transactional Object
+//!
+//! This module provides generic, reusable primitives for persisting data
+//! as objects in a durable store, with optimistic concurrency control and optional
+//! epoch-based fencing.
+//!
+//! Core types
+//! ----------
+//! - `TransactionalObjectOps<T, Id>`: A trait that defines how a transactional object should
```

Comment:
Nit: The *Ops naming suffix is confusing to me.. Is this implying a mix-in style of usage? Can't really think of a better suffix but wanted to make note of it.

Reply:
Not really mix-in. I wanted 2 levels of abstraction:
1. `TransactionalObject` defines an api for users of transactional objects to read and write them
2. Under the hood, a transactional object is implemented using some protocol on some data storage system. We could support a variety of protocols (e.g. S3 sequence files vs direct CAS of a single file) and a variety of storage systems (e.g. S3 vs an rdbms). Each protocol needs to support some primitive operations like read an object and give me its id, write an object at a given id, etc. to support `TransactionalObject`. This trait is meant to define what those operations are so we can swap out different protocols/storage systems

Reply:
I like `TransactionalStorageProtocol` - I'll use that.